src/components/TopBar.test.ts

```diff
@@ -0,0 +1,72 @@
+import { screen } from '@testing-library/react';
+import { h } from 'react-hyperscript-helpers';
+import TopBar from 'src/components/TopBar';
+import { AuthState, authStore } from 'src/libs/state';
+import { renderWithAppContexts as render } from 'src/testing/test-utils';
+
+type AjaxExports = typeof import('src/libs/ajax');
+jest.mock('src/libs/ajax', (): Partial<AjaxExports> => {
+  return {
+    Ajax: jest.fn(),
+  };
+});
+
+jest.mock('src/auth/auth');
+
+jest.mock('react-notifications-component', () => {
+  return {
+    Store: {
+      addNotification: jest.fn(),
+      removeNotification: jest.fn(),
+    },
+  };
+});
+
+type NavExports = typeof import('src/libs/nav');
+jest.mock(
+  'src/libs/nav',
+  (): NavExports => ({
+    ...jest.requireActual<NavExports>('src/libs/nav'),
+    getLink: jest.fn(),
+  })
+);
+
+const itDoesNotShowTheSignInButton = (newState: Partial<AuthState>) => {
+  it('does not display the Sign In button in the menu', () => {
+    // Arrange
+    authStore.update((state) => ({ ...state, ...newState }));
+
+    // Act
+    render(h(TopBar));
+
+    // Assert
+    expect(screen.queryByText('Sign In')).not.toBeInTheDocument();
+  });
+};
+
+const itShowsTheSignInButton = (newState: Partial<AuthState>) => {
+  it('displays the Sign In button in the menu', () => {
+    // Arrange
+    authStore.update((state) => ({ ...state, ...newState }));
+
+    // Act
+    render(h(TopBar));
+
+    // Assert
+    expect(screen.queryByText('Sign In')).not.toBeInTheDocument();
```

Comment:
This case appears to be the same as the last one… it is verifying that `Sign In` is not in the document, which isn't supposed to be true.